package/src/frontend/documentHiddenState.js

```diff
@@ -0,0 +1,27 @@
+
+let isEventAdded = false;
+let callbacks = [];
+
+let muteInBackground = () => {  
+  callbacks.forEach((cb)=>{    
+    if (cb instanceof Function) {
+      cb(document.visibilityState);
+    }
+  })
+};
+
+export function documentHiddenState(callback){
+  callbacks.push(callback);
+
+  if (!isEventAdded) {
+    isEventAdded = true;
+    document.addEventListener('visibilitychange', muteInBackground, false);
+  }
+
+  return {
+    visibilityState: document.visibilityState,
```

Comment:
So far, we don't use the returned `visibilityState` anywhere. Since `document.visibilityState` is globally available, users can always use that one. Since values between returned `visibilityState` and `document.visibilityState` can diverge, let's not return `document.visibilityState`'s current value here